packages/mobile/src/screens/chat-screen/LinkPreview.tsx

```diff
@@ -0,0 +1,160 @@
+import { makeStyles } from 'app/styles'
+
+import { REACTION_LONGPRESS_DELAY } from './constants'
+
+const useStyles = makeStyles(({ spacing, palette, typography }) => ({
+  root: {
+    backgroundColor: palette.white,
+    borderTopLeftRadius: spacing(3),
+    borderTopRightRadius: spacing(3),
+    overflow: 'hidden'
+  },
+  rootIsLinkPreviewOnly: {
+    borderRadius: spacing(3)
+  },
+  thumbnail: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    width: '100%',
+    maxHeight: spacing(50)
+  },
+  image: {
+    display: 'flex',
+    flexGrow: 1,
+    width: '100%',
+    height: '100%'
+  },
+  domainContainer: {
+    borderBottomColor: palette.neutralLight7,
+    borderBottomWidth: 1,
+    backgroundColor: palette.white
+  },
+  domain: {
+    paddingHorizontal: spacing(4),
+    paddingVertical: spacing(1),
+    fontSize: typography.fontSize.small,
+    lineHeight: typography.fontSize.small * 1.5,
+    color: palette.neutral
+  },
+  textContainer: {
+    display: 'flex',
+    marginTop: spacing(4),
+    marginHorizontal: spacing(4),
+    backgroundColor: palette.white
+  },
+  title: {
+    fontSize: typography.fontSize.medium,
+    lineHeight: typography.fontSize.medium * 1.5,
+    fontFamily: typography.fontByWeight.bold
+  },
+  description: {
+    fontSize: typography.fontSize.medium,
+    lineHeight: typography.fontSize.medium * 1.5,
+    marginBottom: spacing(4)
+  },
+  noDescriptionMarginBottom: {
+    marginBottom: spacing(4)
+  }
+}))
+
+type LinkPreviewProps = {
+  href: string
+  isLinkPreviewOnly: boolean
+  onLongPress: (event: GestureResponderEvent) => void
+}
+
+type UnfurlResponse = {
+  url: string
+  url_type: string
+  site_name: string
+  title: string
+  description: string
+  image: string
+  html: string
+  favicon: string
+}
+
+export const LinkPreview = (props: LinkPreviewProps) => {
+  const styles = useStyles()
+  const { href, isLinkPreviewOnly, onLongPress } = props
+
+  const [metadata, setMetadata] = useState<Partial<UnfurlResponse>>()
+  const domain = metadata?.url ? new URL(metadata?.url).hostname : ''
+
+  useEffect(() => {
+    const fn = async () => {
+      try {
+        const sdk = await audiusSdk()
+        const unfurled = await sdk.chats.unfurl({ urls: [href] })
+        setMetadata(unfurled[0])
+      } catch (e) {
+        console.error('Failed to unfurl url', href, e)
+      }
+    }
+    fn()
+  }, [setMetadata, href])
+
+  if (!metadata) {
+    return null
+  }
+
+  return (
+    <Link
+      url={href}
+      delayLongPress={REACTION_LONGPRESS_DELAY}
+      onLongPress={onLongPress}
+    >
+      <View
+        style={[
+          styles.root,
+          isLinkPreviewOnly ? styles.rootIsLinkPreviewOnly : null
+        ]}
+      >
+        {metadata.description || metadata.title ? (
+          <>
+            {metadata.image ? (
+              <View style={styles.thumbnail}>
+                <Image
+                  source={{ uri: metadata.image }}
```

Comment:
is it possible that `metadata.image` doesn't exist here? asking because i see a check for it on line 148

Reply:
We check for it on line 123 so it should exist here. This is the branch for where `description || title` exist, and then on line 148 is the case where they don't exist, so we check for `image` again.